crates/matrix-sdk-ui/src/timeline/controller/state_transaction.rs

```diff
@@ -393,6 +393,7 @@ impl<'a> TimelineStateTransaction<'a> {
         room_data_provider: &P,
         settings: &TimelineSettings,
         event: &AnySyncTimelineEvent,
+        thread_root: &Option<OwnedEventId>,
```

Comment:
nit: it's usually slightly more idiomatic to use `Option<&EventId>` (i.e. move the reference inside the option), since niche optimizations could kick in in this case (and not with `&Option<...>`).